src/classes/OystProduct.php

```diff
@@ -491,15 +491,15 @@ public function addShipment(OystShipment $shipment)
     }
 
     /**
-     * @return string
+     * @return array
      */
     private function getSize()
     {
-        return $this->size;
+        return $this->size ? $this->size->toArray() : array();
```

Comment:
Just to know but why the object is not returned instead of this array ?
If it's for the API part, I think it's better to get the object during the process and then call the `toArray` when you are filling  the information for the product.